test/controllers/changesets_controller_test.rb

```diff
@@ -121,8 +125,9 @@ def test_index_bbox
   # Checks the display of the user changesets listing
   def test_index_user
     user = create(:user)
-    create(:changeset, :user => user)
-    create(:changeset, :closed, :user => user)
+    create(:changeset, :user => user, :num_changes => 1)
+    create(:changeset, :closed, :user => user, :num_changes => 1)
+    user.reload
```

Comment:
Why do we need to reload the user? Is that to cause the cached changeset count on the user to update or something?

Reply:
This test was comparing empty list of changesets (because changesets without changes are filtered out) with another empty list of changesets (because `user.changesets` weren't updated after `create(:changeset, ...)`). So it wasn't testing anything and it wasn't testing my changes.